e2e_tests/src/e2e_conduit/fixtures/importers/follower_algod.py

```diff
@@ -0,0 +1,115 @@
+import atexit
+import glob
+import json
+import logging
+import os
+import shutil
+import tempfile
+
+import boto3
+from botocore.config import Config
+from botocore import UNSIGNED
+
+from e2e_common.util import hassuffix, xrun, firstFromS3Prefix, countblocks, atexitrun
+from e2e_conduit.fixtures.importers.importer_plugin import ImporterPlugin
+
+logger = logging.getLogger(__name__)
+
+
+class FollowerAlgodImporter(ImporterPlugin):
+    def __init__(self, sourcenet):
+        self.algoddir = None
+        self.sourcenet = sourcenet
+        self.last = None
+        super().__init__()
+
+    @property
+    def name(self):
+        return "algod_follower"
+
+    @property
+    def lastblock(self):
+        if self.last is None:
+            raise RuntimeError("algod importer has no blockfiles configured")
+        return self.last
+
+    def resolve_config_input(self):
+        with open(os.path.join(self.algoddir, "algod.net"), "r") as algod_net:
+            self.config_input["netaddr"] = "http://" + algod_net.read().strip()
+        with open(os.path.join(self.algoddir, "algod.token"), "r") as algod_token:
+            self.config_input["token"] = algod_token.read().strip()
+
+    def resolve_config_output(self):
+        self.config_output["algod_token"] = self.config_input["token"]
+        self.config_output["algod_net"] = self.config_input["netaddr"]
+        self.config_output["algod_data_dir"] = self.algoddir
+
+    def setup(self, accumulated_config):
+        try:
+            conduit_dir = accumulated_config["conduit_dir"]
+        except KeyError as exc:
+            logger.error(
+                f"AlgodImporter needs to be provided with the proper config: {exc}"
+            )
+            raise
+        source_is_tar = hassuffix(
+            self.sourcenet, ".tar", ".tar.gz", ".tar.bz2", ".tar.xz"
+        )
+        if not (source_is_tar or (self.sourcenet and os.path.isdir(self.sourcenet))):
+            # Assuming self.sourcenet is an s3 source net
+            tarname = f"{self.sourcenet}.tar.bz2"
+            # fetch test data from S3
+            bucket = "algorand-testdata"
+
+            s3 = boto3.client("s3", config=Config(signature_version=UNSIGNED))
+            prefix = "indexer/e2e4"
+            if "/" in tarname:
+                cmhash_tarnme = tarname.split("/")
+                cmhash = cmhash_tarnme[0]
+                tarname =cmhash_tarnme[1]
+                prefix+="/"+cmhash
+                tarpath = os.path.join(conduit_dir, tarname)
+            else:
+                tarpath = os.path.join(conduit_dir, tarname)
+            success = firstFromS3Prefix(s3, bucket, prefix, tarname, outpath=tarpath)
+            if not success:
+                raise Exception(
+                    f"failed to locate tarname={tarname} from AWS S3 path {bucket}/{prefix}"
+                )
+            source_is_tar = True
+            self.sourcenet = tarpath
```

Comment:
This seems problematic. Can follow mode support a predefined ledger?

Reply:
Kind of...The problem I had w/ the existing e2e setup was that when you run `goal network start` with predefined data you get two nodes started at round 40 (or however many rounds there are). 

This wasn't a problem when all we wanted was the block data since you can retrieve any round, but in order to get the delta for those rounds we need to start at 0 and progress to 40. 

Potentially in the future we can add some persistence to deltas so that you can start a network w/ a larger number of deltas stored instead of having to populate the cache round by round.

Reply:
Sounds like a larger discussion is needed to figure out whether or not we're outgrowing the e2e artifact.

Thinking out loud though... I think we could take the existing network snapshot and attach a follower node to it. You'd have to pass in a custom "phonebook" to the follow node so that it caught up from the test node.

Reply:
I was mistaken -- what you have here looks great.